main.js

```diff
@@ -25,17 +25,32 @@ function getNewHexes() {
     currentHexes = [];
     var newColor;
     for(i = 0; i < mainColorBoxes.length; i++) {
-        newColor = getRandomHex().toUpperCase();
-        mainColorBoxes[i].firstElementChild.style.backgroundColor = `#${newColor}`;
-        mainColorBoxes[i].lastElementChild.innerText = `#${newColor}`;
+        var thisColorBoxLock = mainColorBoxes[i].firstElementChild.firstElementChild
+        if(thisColorBoxLock.classList.contains('unlocked')) {
+            newColor = getRandomHex().toUpperCase();
+            mainColorBoxes[i].firstElementChild.style.backgroundColor = `#${newColor}`;
+            mainColorBoxes[i].lastElementChild.innerText = `#${newColor}`;
+        }
     }
 }
 
 function lockToggle(event) {
     if (event.getAttribute('src') === './assets/unlocked.png') {
-        event.src = './assets/locked.png'
+        event.src = './assets/locked.png';
+        toggleLockClass(event);
     } else {
-        event.src = './assets/unlocked.png'
+        event.src = './assets/unlocked.png';
+        toggleLockClass(event);
+    }
+}
+
+function toggleLockClass(element) {
+    if(element.classList.contains('locked')) {
+        element.classList.remove('locked');
+        element.classList.add('unlocked');
+    } else {
+        element.classList.add('locked');
+        element.classList.remove('unlocked');
     }
 }
```

Comment:
We can refactor this function to remove the locked class since it's not targeted in the rest of the code. The conditional statement in the `getNewHexes()` function is only targeting the unlocked class. 
you can use `add` and `remove` like you were using or use the `toggle` method.